File: leetcode-04/solutions/python3/0073.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        m = len(matrix)
        n = len(matrix[0])

        isFirstRow = False
        isFirstCol = False

        for j in range(n):
            if matrix[0][j] == 0:
                isFirstRow = True

        for i in range(m):
            if matrix[i][0] == 0:
                isFirstCol = True

        for i in range(1, m):
            for j in range(1, n):
                if matrix[i][j] == 0:
                    matrix[i][0] = 0
                    matrix[0][j] = 0

        for i in range(1, m):
            for j in range(1, n):
                if matrix[i][0] == 0 or matrix[0][j] == 0:
                    matrix[i][j] = 0

        if isFirstRow:
            for j in range(n):
                matrix[0][j] = 0

        if isFirstCol:
            for i in range(m):
                matrix[i][0] = 0

File: leetcode-04/solutions/cpp/0073.cpp
class Solution {
 public:
  void setZeroes(vector<vector<int>>& matrix) {
    const int m = matrix.size();
    const int n = matrix[0].size();

    bool isFirstRow = false;
    bool isFirstCol = false;

    for (int j = 0; j < n; j++)
      if (matrix[0][j] == 0) isFirstRow = true;

    for (int i = 0; i < m; i++)
      if (matrix[i][0] == 0) isFirstCol = true;

    for (int i = 1; i < m; i++)
      for (int j = 1; j < n; j++)
        if (matrix[i][j] == 0) {
          matrix[i][0] = 0;
          matrix[0][j] = 0;
        }

    for (int i = 1; i < m; i++)
      for (int j = 1; j < n; j++)
        if (matrix[i][0] == 0 || matrix[0][j] == 0) matrix[i][j] = 0;

    if (isFirstRow)
      for (int j = 0; j < n; j++) matrix[0][j] = 0;

    if (isFirstCol)
      for (int i = 0; i < m; i++) matrix[i][0] = 0;
  }
};